src/config/parse.c

```diff
@@ -873,302 +969,491 @@ configParse(unsigned int argListSize, const char *argList[], bool resetLogLevel)
+                        {
+                            dependResolved = cfgDefOptionDependValueValid(config->command, optionId, strZ(varStr(dependValue)));
 
-                            for (unsigned int listIdx = 0;
-                                    listIdx < cfgDefOptionDependValueTotal(commandId, optionDefId); listIdx++)
+                            // If depend not resolved and option value is set on the command-line then error.  It's OK to have
+                            // unresolved options in the config file because they may be there for another command.  For instance,
+                            // spool-path is only loaded for the archive-push command when archive-async=y, and the presence of
+                            // spool-path in the config file should not cause an error here, it will just end up null.
+                            if (!dependResolved && optionSet && parseOptionValue->source == cfgSourceParam)
                             {
-                                const char *dependValue = cfgDefOptionDependValue(commandId, optionDefId, listIdx);
+                                // Get the depend option name
+                                const String *dependOptionName = STR(cfgOptionKeyIdxName(dependOptionId, optionKeyIdx));
 
-                                // Build list based on depend option type
-                                if (dependOptionDefType == cfgDefOptTypeBoolean)
-                                {
-                                    // Boolean outputs depend option name as no-* when false
-                                    if (strcmp(dependValue, "0") == 0)
-                                        dependOptionName = strNewFmt("no-%s", cfgOptionName(dependOptionId));
-                                }
-                                else
+                                // Build the list of possible depend values
+                                StringList *dependValueList = strLstNew();
+
+                                for (unsigned int listIdx = 0;
+                                        listIdx < cfgDefOptionDependValueTotal(config->command, optionId); listIdx++)
                                 {
-                                    ASSERT(dependOptionDefType == cfgDefOptTypePath || dependOptionDefType == cfgDefOptTypeString);
-                                    strLstAdd(dependValueList, strNewFmt("'%s'", dependValue));
+                                    const char *dependValue = cfgDefOptionDependValue(config->command, optionId, listIdx);
+
+                                    // Build list based on depend option type
+                                    if (dependOptionDefType == cfgDefOptTypeBoolean)
+                                    {
+                                        // Boolean outputs depend option name as no-* when false
+                                        if (strcmp(dependValue, "0") == 0)
```

Comment:
Yeah, that makes sense, and also where the default is set at line 1389.

But we should really use `ZERO_Z` since it is already defined, rather than creating a new define. This code is far older than the stringz.h header that contains `ZERO_Z`.

Reply:
I'll take a look at resolving this...

Reply:
Done in 1c66629f61689fa9.